sdks/python/apache_beam/dataframe/io.py

```diff
@@ -521,10 +535,15 @@ def create_tracker(self, restriction):
       return beam.io.restriction_trackers.UnsplittableRestrictionTracker(
           tracker)
 
-  def process(self, readable_file, tracker=beam.DoFn.RestrictionParam()):
+  def process(
+      self, readable_file, path_indices, tracker=beam.DoFn.RestrictionParam()):
     reader = self.reader
     if isinstance(reader, str):
       reader = getattr(pd, self.reader)
+    start_index = (
+        tracker.current_restriction().start +
+        path_indices[readable_file.metadata.path] *
+        10**int(math.log(2**64 // len(path_indices), 10)))
```

Comment:
IIUC, this finds the order of magnitude of `2**64 // len(path_indices)`. So it could be substantially smaller than the largest file we can support (e.g. 9999 -> 1000).

Should there be some logic to handle the case where a file is too large and will produce collisions? I'm not sure what we could do in that case though.

Reply:
This becomes an error in the exabyte range (e.g. millions of TB-sized files). That's probably a ways off, but just in case I added a check.

Reply:
That's assuming a reasonably uniform distribution of file sizes though. Then again, millions of small files and a small number of TB-sized file is probably an equally remote possibility.

Reply:
Yeah, the unequal file size distributions is a possible case. Still quite unlikely, but at least now they'll get a check.